### ISSUE TITLE
losses are always nan

### ISSUE THREAD

**Issue by m-ali-awan** [ML4ITS/mtad-gat-pytorch#12]
Hi, hope you are fine.
Thanks for this wonderful work.
I tried training with MSL, and SMD, and my losses are always nan.
Moreover, I also tried GDN repo, and I found that there is a difference in MSL data as compared to this repo.
Thanks for any help.

Regards,
Ali

**Comment by m-ali-awan** [comment]
I have further dived into code and came to know that outputs from TemporalAttentionLayer are nan always, and from FeatureAttentionLayer some come out as nans.

**Comment by axeloh** [comment]
If you haven't already, I suggest you ensure that
1. input data has the correct format 
2. data itself does not contain nan and are scaled properly before input to the model.

Regards,
Axel

**Comment by ylic204** [comment]
> I have further dived into code and came to know that outputs from TemporalAttentionLayer are nan always, and from FeatureAttentionLayer some come out as nans.

Have you solved this problem yet? I also have this problem, During training, the losses are all nan,and there are many zeros in the data. And I don't understand why the num_values in labeled_anomalies.csv and the shape in the .npy file in the train folder are different. For example, C-1 in labeled_anomalies.csv is 2264, but C-1.npy is 2158. '2264' and '2158' don't match.

**Comment by JinYang88** [comment]
Same problem with the SMD dataset, using the default hyperparameters.

**Comment by JinYang88** [comment]
Setting use_gatv2 to False can produce normal loss.

**Comment by fga-telenor** [comment]
Check out my answer on #13, I believe it is due to uninitialized bias parameters.

**Comment by JinYang88** [comment]
> Check out my answer on #13, I believe it is due to uninitialized bias parameters.

It works, great!
